bentoml/_internal/resource.py

```diff
@@ -238,22 +245,24 @@ def from_system(cls) -> int:
         try:
             pynvml.nvmlInit()
             device_count = pynvml.nvmlDeviceGetCount()
-            return device_count
+            return list(range(device_count))
         except (pynvml.nvml.NVMLError, OSError):
-            return 0
+            return []
         finally:
             try:
                 pynvml.nvmlShutdown()
             except Exception:  # pylint: disable=broad-except
                 pass
 
     @classmethod
-    def validate(cls, val: int):
-        if val < -1:
-            raise BentoMLConfigException(f"Invalid negative GPU resource limit {val}.")
-        if val > cls.from_system():
+    def validate(cls, val: t.List[int]):
+        if any([gpu_index < -1 for gpu_index in val]):
```

Comment:
Is -1 a valid device ID?

Reply:
from my understanding, the -1 flag is to disable access to GPUs altogether. @sauyon can you confirm if that is indeed correct?

Reply:
I don't think we should support it as an element of the list even if it is; we should just parse -1 into an empty list.

Reply:
got it. updated the logic and changed the tests accordingly